
Load the dataset and relevant R pacakges

```{r eval=FALSE}

# conda activate cicero
library(Seurat)
library(Signac)
library(tidyverse)
library(viridis)
library(cowplot)
library(ggrepel)
library(GenomicRanges)
theme_set(theme_cowplot())

setwd('~/swaruplab/smorabit/analysis/PiD_2021/intro')

data_dir <- 'data/'
fig_dir <- 'figures/'

# load the PiD + AD seurat_obj object
seurat_obj <- readRDS(file='/dfs7/swaruplab/smorabit/analysis/PiD_2021/data/PiD_AD_integrated_in_progress_11-01-21.rds')


# update the fragments path:

f <- Fragments(seurat_obj)[[1]]
head(f@cells)

# update fragments
samples <- unique(seurat_obj$Sample)
frag_list <- list()
for(i in 1:length(samples)){
  cur_sample <- samples[[i]]
  print(cur_sample)
  cells <- subset(seurat_obj@meta.data, Sample == cur_sample) %>% rownames

  bcs <- do.call(rbind, strsplit(cells, '_'))[, length(strsplit(cur_sample, '_')[[1]]) + 1]
  names(bcs) <- cells

  frag_list[[i]] <- CreateFragmentObject(
    path = paste0("/dfs7/swaruplab/smorabit/collab/AMRF/analysis/ATAC/cellRanger_Fragments/", cur_sample, "/outs/fragments.tsv.gz"),
    cells = bcs,
    validate.fragments=FALSE
  )
}
Fragments(seurat_obj) <- NULL
Fragments(seurat_obj) <- frag_list






peaks <- read.csv('~/swaruplab/smorabit/analysis/PiD_2021/data/peak_table.csv')

peaks <- peaks[,1:3]
write.table(peaks, quote=FALSE, row.name=FALSE, col.name=FALSE, sep='\t', file='~/swaruplab/smorabit/analysis/PiD_2021/data/pid_peaks.bed')


# fix some of the meta-data that was different between the two studies
seurat_obj$Diagnosis <- ifelse(seurat_obj$Dataset == 'AD', seurat_obj$Diagnosis, seurat_obj$DX)
seurat_obj$Diagnosis <- ifelse(seurat_obj$Diagnosis == "Pick's disease", 'PiD', seurat_obj$Diagnosis)


# add the UMAP coordinates:
seurat_obj$UMAP_1 <- seurat_obj@reductions$umap@cell.embeddings[,1]
seurat_obj$UMAP_2 <- seurat_obj@reductions$umap@cell.embeddings[,2]



# Diagnosis
# Sex
# Age
seurat_obj@meta.data %>% subset(Diagnosis == "PiD") %>% .$SampleID %>% table
seurat_obj@meta.data %>% subset(Diagnosis == "PiD") %>% .$celltype %>% table
seurat_obj@meta.data %>% subset(Diagnosis == "PiD") %>% .$cluster_name %>% table
seurat_obj@meta.data %>% subset(Dataset == "PiD") %>% .$Diagnosis %>% table
seurat_obj@meta.data %>% subset(Dataset == "AD") %>% .$Diagnosis %>% table
seurat_obj$celltype %>% table

umap_theme <- theme(
  axis.line=element_blank(),
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
  axis.title.x=element_blank(),
  axis.title.y=element_blank(),
  panel.background=element_blank(),
  panel.border=element_blank(),
  panel.grid.major=element_blank(),
  panel.grid.minor=element_blank(),
  plot.background=element_blank(),
  plot.title = element_text(hjust = 0.5)
)

load('/dfs7/swaruplab/smorabit/analysis/AD_NucSeq_2019/batch_correction/liger/update/celltype-analysis/data/color_scheme.rda')
load('~/swaruplab/smorabit/analysis/AD_NucSeq_2019/atac_analysis/all_data/celltype-analysis/data/color_scheme.rda')

color_scheme_snATAC_celltype$EX <- 'turquoise'

```

QC vln plots

```{r eval=FALSE}


seurat_obj <- subset(seurat_obj, Dataset == 'PiD')


p <- VlnPlot(seurat_obj, group.by='SampleID', features = 'nCount_peaks', pt.size=0) +
  geom_boxplot(outlier.shape=NA, width=.25, fill='white') +
  scale_y_continuous(labels=scales::comma) +
  xlab('') + NoLegend() + ggtitle('')

pdf(paste0(fig_dir, 'vln_nCount_peaks_samples_pid.pdf'), height=4, width=8)
p
dev.off()

p <- VlnPlot(seurat_obj, group.by='SampleID', features = 'TSSEnrichment', pt.size=0) +
  geom_boxplot(outlier.shape=NA, width=.25, fill='white') +
  scale_y_continuous(labels=scales::comma) +
  xlab('') + NoLegend() + ggtitle('')

pdf(paste0(fig_dir, 'vln_TSS_samples_pid.pdf'), height=4, width=8)
p
dev.off()


```

Plot UMAPs for different variables:

```{r eval=FALSE}

################################################################################
# UMAP for clusters
################################################################################
# get cluster centroids:
centroid_df <- data.frame()
for(cur_cluster in unique(seurat_obj$cluster_name)){

  cur_meta <- subset(seurat_obj@meta.data, cluster_name == cur_cluster)

  # average values of UMAP_1 and UMAP_2
  df <- data.frame(
    cluster = cur_cluster,
    UMAP_1 = mean(cur_meta$UMAP_1),
    UMAP_2 = mean(cur_meta$UMAP_2)
  )
  centroid_df <- rbind(centroid_df, df)
}

p1 <- seurat_obj@meta.data %>% ggplot(aes(x=UMAP_1, y=UMAP_2, color=cluster_name)) +
  geom_point(size=0.5, alpha=0.5) +
  geom_text_repel(data=centroid_df, aes(x=UMAP_1, y=UMAP_2), label=centroid_df$cluster, color='black') +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend()


pdf(paste0(fig_dir, 'umap_clusters.pdf'), width=6, height=6, useDingbats=FALSE)
p1
dev.off()

p1 <- seurat_obj@meta.data %>% ggplot(aes(x=UMAP_1, y=UMAP_2, color=cluster_name)) +
  geom_point(size=0.5, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend()

png(paste0(fig_dir, 'umap_clusters_no_title.png'), width=6, height=6, units='in', res=500)
p1 + theme(strip.background=element_blank())
dev.off()


################################################################################
# UMAP for cell types
################################################################################

p1 <- seurat_obj@meta.data %>% ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  geom_point(size=0.5, alpha=0.5) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  umap_theme + NoLegend()


png(paste0(fig_dir, 'umap_celltype.png'), width=6, height=6, res=600, units='in')
p1 + theme(strip.background=element_blank())
dev.off()


################################################################################
# UMAP for cell types split by Diagnosis with title
################################################################################

# PiD Control cells
p1 <- seurat_obj@meta.data %>% subset(Dataset == 'PiD' & Diagnosis == 'Control') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'PiD' | Diagnosis != 'Control'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('Control (PiD)')


# PiD Disease cells
p2 <- seurat_obj@meta.data %>% subset(Dataset == 'PiD' & Diagnosis == 'PiD') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'PiD' | Diagnosis != 'PiD'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('Disease (PiD)')

# AD Control cells
p3 <- seurat_obj@meta.data %>% subset(Dataset == 'AD' & Diagnosis == 'Control') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'AD' | Diagnosis != 'Control'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('Control (AD)')


# AD Disease cells
p4 <- seurat_obj@meta.data %>% subset(Dataset == 'AD' & Diagnosis == 'AD') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'AD' | Diagnosis != 'AD'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('Disease (AD)')


png(paste0(fig_dir, 'umap_Diagnosis.png'), width=10, height=10, res=500, units='in')
(p2 + p4) /
(p1 + p3)
dev.off()


################################################################################
# UMAP for cell types split by Diagnosis without title
################################################################################

# PiD Control cells
p1 <- seurat_obj@meta.data %>% subset(Dataset == 'PiD' & Diagnosis == 'Control') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'PiD' | Diagnosis != 'Control'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('')


# PiD Disease cells
p2 <- seurat_obj@meta.data %>% subset(Dataset == 'PiD' & Diagnosis == 'PiD') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'PiD' | Diagnosis != 'PiD'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('')

# AD Control cells
p3 <- seurat_obj@meta.data %>% subset(Dataset == 'AD' & Diagnosis == 'Control') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'AD' | Diagnosis != 'Control'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('')


# AD Disease cells
p4 <- seurat_obj@meta.data %>% subset(Dataset == 'AD' & Diagnosis == 'AD') %>%
  ggplot(aes(x=UMAP_1, y=UMAP_2, color=celltype)) +
  scale_color_manual(values=color_scheme_snATAC_celltype) +
  geom_point(
    data=seurat_obj@meta.data %>% subset(Dataset != 'AD' | Diagnosis != 'AD'),
    aes(x=UMAP_1, y=UMAP_2), color='lightgray', alpha=0.5, size=0.25
  ) +
  geom_point(size=1, alpha=0.5) +
  # scale_color_manual(values=color.scheme) +
  umap_theme + NoLegend() + ggtitle('')


png(paste0(fig_dir, 'umap_Diagnosis_no_title.png'), width=3, height=12, res=500, units='in')
p2 / p4 / p1 / p3
dev.off()


```

Make a coverage plot for cell-type marker genes:

```{r eval=FALSE}

# load marker genes from NatGen study
load('/dfs7/swaruplab/smorabit/analysis/AD_NucSeq_2019/batch_correction/liger/update/data/all_DEGs.rda')

top_markers <- celltype.markers %>% group_by(cluster) %>% top_n(20, wt=avg_logFC)
top_markers %>% subset(cluster == 'PER.END')
cur_markers <- top_markers %>% subset(cluster == 'ASC')
cur_markers <- top_markers %>% subset(cluster == 'EX')
cur_markers <- top_markers %>% subset(cluster == 'MG')
cur_markers <- top_markers %>% subset(cluster == 'PER.END')
cur_markers <- top_markers %>% subset(cluster == 'INH')


cur_markers$region_length <- sapply(cur_markers$gene, function(x){
  Annotation(seurat_obj) %>% subset(gene_name == x) %>% width %>% max
})
cur_markers

DefaultAssay(seurat_obj) <- 'peaks'

# pericyte / endothelial markers:
cur_gene <- 'FLT1' # looks good but gene is hella long

# astro markers
cur_gene <- 'RFX4'
cur_gene <- 'GFAP'

# Micro markers
cur_gene <- 'CX3CR1'
cur_gene <- 'CSF1R'

# ODC markers
cur_gene <- 'MOG'
cur_gene <- 'OPALIN'

# ODC markers
cur_gene <- 'PDGFRA'
cur_gene <- 'BCAN'
cur_gene <- 'VCAN'


# neuron marker
cur_gene <- 'SYT1' # bad
cur_gene <- 'SNAP25' # bad
cur_gene <- 'MAP2' # bad

# EX markers:
cur_gene <- 'RALYL'
cur_gene <- 'RORB'

# INH markers:
cur_gene <- 'KCNC2' # okay
cur_gene <- 'VIP' # also okay
cur_gene <- 'SST' # really good
# VIP PVALB SST LAMP5

cur_region <- Annotation(seurat_obj) %>% subset(gene_name == cur_gene)
cur_chr <- unique(as.character(seqnames(cur_region)))
plot_region <- paste(cur_chr, min(start(cur_region)),max(end(cur_region)), sep='-')
width(cur_region) %>% max


cov_plot1 <- Signac::CoveragePlot(
  seurat_obj,
  region=cur_gene,
  group.by='celltype',
  extend.upstream=1000,
  extend.downstream=1000,
  peaks=FALSE
)

pdf(paste0(fig_dir, 'coverage_plots/', cur_gene, "_coverageplot_celltype.pdf"), height=10, width=5)
cov_plot1
dev.off()



```


Example motif feature plot / violin plot :

```{r eval=FALSE}

# motif featureplot
DefaultAssay(seurat_obj) <- 'peaks'
motif_names <- GetMotifData(seurat_obj, slot='motif.names')

DefaultAssay(seurat_obj) <- 'chromvar'
cur_motif <- 'SPI1'
cur_motif_ID <- names(motif_names[grepl(cur_motif, motif_names)])


p <- FeaturePlot(seurat_obj, split.by='Diagnosis', features=cur_motif_ID, order=FALSE , raster=FALSE) +
scale_color_gradient2(low=scales::muted('blue'), mid='white', high=scales::muted('red')) +
theme(plot.margin = unit(c(0, 0, 0, 0), "in")) + umap_theme +
ggtitle(paste0(cur_motif, ' motif enrichment'))

comparisons <- list(c(1,2),  c(1,3), c(2,3))


p <- VlnPlot(
    seurat_obj %>% subset(Dataset=='PiD'),
    features=cur_motif_ID,
    pt.size=0, split.by='Diagnosis', group.by='celltype') +
  stat_compare_means(method='wilcox.test', label='p.signif', label.y=3) +
  geom_hline(yintercept = 0, linetype='dashed')


pdf(paste0(fig_dir, cur_motif, '_enrichment.pdf'), width=12, height=4, useDingbats=FALSE)
p
dev.off()



```

One differential test:

```{r eval=FALSE}


x1 <- GetAssayData(seurat_obj, assa='peaks', slot='counts')
x2 <- GetAssayData(seurat_obj, assa='peaks', slot='data')

sum(is.na(seurat_obj@meta.data$Age))

cur_seurat <- subset(seurat_obj, celltype == 'MG' & Diagnosis != 'Control')
Idents(cur_seurat) <- cur_seurat$Diagnosis

test <- FindMarkers(
  cur_seurat,
  ident.1 = 'AD',
  ident.2 = 'PiD',
  assay = 'peaks',
  test.use = 'LR',
  min.pct=0,
  logfc.threshold=0.1,
  only.pos=FALSE,
  latent.vars = c('nCount_peaks', 'Sex', 'Batch', 'Age')
)

```
