

In this notebook we take .bam files from cell ranger, and the fully processed
Seurat object, to make a bigwig track for each cell group that we can vive
on the UCSC genome browser.

# 1: Load Data

```{r eval=FALSE}
# conda activate spatial

library(Seurat)
library(tidyverse)

# load seurat obj:
seurat_obj <- readRDS(file='/dfs7/swaruplab/smorabit/analysis/PiD_2021/data/PiD_AD_integrated.rds')

# subset just PiD:
seurat_obj <- subset(seurat_obj, Dataset == 'PiD')

setwd('/dfs3b/swaruplab/smorabit/analysis/PiD_2021/trackhubs')

# write the peaks to a bed file:


```

# Step 1: Get Barcode lists for each cell type in each sample
For each sample, make a .tsv file that tells sinto how to split up the .bams

```{r eval=FALSE}

# IMPORTANT:
# cellranger aggr doesn't make a bam file, so we have to use the ones from celranger count.
# This means that the barcode suffix on each reads for each file have -1 at the end for all
# of the files, regardless what it says in the Seurat object.

################################################################################
# split by celltype
################################################################################

dir.create('data/barcodes/')


samples <- unique(seurat_obj$SampleID) %>% as.character

# loop over each sample
for(sample in samples){
  print(sample)

  # print cell barcodes for one whole Sample:
  barcodes <- seurat_obj@meta.data %>%
    subset(Sample == sample)

  bcodes <- rownames(barcodes)
  if(sample %in% c("Sample1", "Sample2", "Sample3", "Sample4")){
    bcodes <- do.call(rbind, strsplit(bcodes, '_'))[,2]
  } else{
    bcodes <- do.call(rbind, strsplit(bcodes, '_'))[,3]
  }
  bcodes <- do.call(rbind, strsplit(bcodes, '-'))[,1]

  # add cell type column
  out_df <- data.frame(
    'barcode'=paste0(bcodes, '-1'),
    'file' = gsub(' ', '_', barcodes$celltype)
  )

  if(sample %in% c("Sample1", "Sample2", "Sample3", "Sample4")){
    name <- sample
  } else{
    name <- gsub("Sample", "Swarup_Lab", sample)
  }
  print(name)

  # write
  write.table(out_df, file=paste0('data/barcodes/', name, '_celltype_barcodes.tsv'), sep='\t', col.names=FALSE, row.names=FALSE, quote=FALSE)

}

```

# Step 2: Run `sinto_split-bams.sub` script

```{bash}

cd ~/swaruplab/smorabit/analysis/PiD_2021/trackhubs/

mkdir -p data/bams/celltypes/

cd ~/swaruplab/smorabit/analysis/PiD_2021/trackhubs/

# run the script!
sbatch ../bin/sinto_split_bams.sub

```


## Step 3: write list of files to combined using samtools

```{r eval=FALSE}

# cell types
groups <- seurat_obj$celltype  %>% as.character %>% unique
bam_dir <- '/dfs3b/swaruplab/smorabit/analysis/PiD_2021/trackhubs/data/bams/celltypes/'

# loop over all celltypes
samples <- dir(bam_dir)

dir.create('data/bam_merge_lists/')
for(group in groups){
  bam_files <- paste0(bam_dir, samples, '/', group, '.bam')

  # make sure we only look at files that exist! For some small clusters, they may not be
  # present across all samples so a .bam wouldn't be made in that case.
  bam_files <- bam_files[file.exists(bam_files)]

  write.table(bam_files, file=paste0('data/bam_merge_lists/', group, '.tsv'), sep='\t', col.names=FALSE, row.names=FALSE, quote=FALSE)
}

# PiD
all_samples <- samples

sample_df <- seurat_obj@meta.data %>% select(c(DX, SampleID)) %>% distinct()
samples <- c('Sample1', 'Sample3', 'Swarup_Lab_8', 'Swarup_Lab_24', 'Swarup_Lab_17', 'Swarup_Lab_13', 'Swarup_Lab_11')

for(group in groups){
  bam_files <- paste0(bam_dir, samples, '/', group, '.bam')
  bam_files <- bam_files[file.exists(bam_files)]
  write.table(bam_files, file=paste0('data/bam_merge_lists/', group, '_PiD.tsv'), sep='\t', col.names=FALSE, row.names=FALSE, quote=FALSE)
}

# Control
sample_df <- seurat_obj@meta.data %>% select(c(DX, SampleID)) %>% distinct()
samples <- c('Sample2', 'Sample4', 'Swarup_Lab_9', 'Swarup_Lab_6', 'Swarup_Lab_5', 'Swarup_Lab_16', 'Swarup_Lab_15', 'Swarup_Lab_14', 'Swarup_Lab_10')

for(group in groups){
  bam_files <- paste0(bam_dir, samples, '/', group, '.bam')
  bam_files <- bam_files[file.exists(bam_files)]
  write.table(bam_files, file=paste0('data/bam_merge_lists/', group, '_control.tsv'), sep='\t', col.names=FALSE, row.names=FALSE, quote=FALSE)
}

```

## Step 4: Merge bam files based on lists we made in step 3:

```{r eval=FALSE}

cd /dfs3b/swaruplab/smorabit/analysis/PiD_2021/trackhubs/
sbatch ../bin/merge_bams.sub

```

## Step 5: Convert Merged .bams to bigwig format:

```{bash eval=FALSE}

sbatch ../bin/bam_to_bigwig.sub


# re-name .bw to .bigWig
cd data/bigwigs/

for file in ./*.bw
do
  name=$(basename $file .bw)
  mv $file $name.bigWig
done


```



## Step 6: Make necessary files for UCSC browser:

* genomes.txt file
* trackDb.txt file
* hub.txt file


track ODC_i_AD
bigDataUrl ODC_i_AD.bigWig
shortLabel ODC_i_AD
longLabel ODC_i_AD
type bigWig
visibility full
color 255,171,255
autoScale on

```{r eval=FALSE}

load('/dfs3b/swaruplab/smorabit/analysis/AD_NucSeq_2019/batch_correction/liger/update/celltype-analysis/data/color_scheme.rda')



colors <- unlist(color_scheme_snRNA_celltype)

bigwig_dir = "/dfs3b/swaruplab/smorabit/analysis/PiD_2021/trackhubs/data/bigwigs/"
bigwigs <- dir(bigwig_dir)

scale_factor <- 1.6

outfile <- "/dfs3b/swaruplab/smorabit/analysis/PiD_2021/trackhubs/data/trackDb.txt"

for(bw in bigwigs){

  print(bw)

  # get names
  cur_name <- substr(bw, 1, nchar(bw)-7)
  cur_anno <- strsplit(cur_name, '[_]')[[1]][1]

  # get color
  cur_color <- colors[cur_anno]
  cur_rgb <- grDevices::col2rgb(cur_color) %>% as.numeric %>% paste0(collapse=",")

  # get group
  cur_group <- strsplit(bw, '[_]')[[1]][2]
  cur_group <- strsplit(cur_group, '[.]')[[1]][1]

  # how many cells in this group?
  if(cur_group == 'PiD'){
    cur_nCells <- subset(seurat_obj@meta.data, celltype == cur_anno & DX == "Pick's disease") %>% nrow
  } else{
    cur_nCells <- subset(seurat_obj@meta.data, celltype == cur_anno & DX == "Control") %>% nrow
  }

  # scale number of cells for visualiztion
  cur_nCells <- round(cur_nCells * scale_factor)
  cur_view <- paste0("0:", cur_nCells)
  print(cur_view)

  # write all info
  lines <- list(
    paste("track", cur_name),
    paste("bigDataUrl", bw),
    paste("shortLabel", cur_name),
    paste('longLabel', cur_name),
    paste('type bigWig'),
    paste('visibility full'),
    paste('color', cur_rgb),
    paste('autoScale on'),
  #  paste('viewLimits', cur_view),
    paste("")
  )

  for(l in lines){
    cat(l, file = outfile, sep="\n", append=TRUE)
  }

}





```

## Step 7: Moving stuff to the right spot

* Move all bigwigs to artemis
* put them in a web-facing folder such as

/var/www/html/Nurr2c_trackhubs/

* point the UCSC browser to this folder and voila!

Put them here for this paper:
https://swaruplab.bio.uci.edu/Nurr2c_trackhubs/


```{r eval=FALSE}

DA_peaks_clusters_diagnosis.rda

load('~/swaruplab/smorabit/analysis/AD_NucSeq_2019/atac_analysis/all_data/data/DA_peaks/DA_peaks_celltypes_diagnosis.rda')

tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'ASC'); range(tmp$avg_logFC)
tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'ODC'); range(tmp$avg_logFC)
tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'EX'); range(tmp$avg_logFC)
tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'OPC'); range(tmp$avg_logFC)
tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'INH'); range(tmp$avg_logFC)
tmp <- subset(da_peaks, Diagnosis == 'AD' & cluster == 'MG'); range(tmp$avg_logFC)



```
