
```{r eval=FALSE}
# conda activate cicero
library(ArchR)
library(tidyverse)

setwd('~/swaruplab/smorabit/analysis/PiD_2021/with_AD/')

# set archR genome
addArchRGenome("hg38")
addArchRThreads(threads = 8)

```

# create arrow files

```{r eval=FALSE}


cellranger_dir <-  '~/swaruplab/smorabit/data/PiD_2021/March2021/cellranger-count/'
sample_names <- dir(cellranger_dir)[grepl('Swarup_Lab', dir(cellranger_dir))]

# for now exclude samples 7, 18, 23 since they aren't done running:
sample_names <- sample_names[!c(sample_names %in% c('Swarup_Lab_7', 'Swarup_Lab_18', 'Swarup_Lab_23'))]
sample_files <- paste0(cellranger_dir, sample_names, '/outs/fragments.tsv.gz')

sample_names <- gsub('Swarup_Lab', 'Sample', sample_names)

ArrowFiles <- createArrowFiles(
  inputFiles = sample_files,
  sampleNames = sample_names,
  minTSS = 4, #Dont set this too high because you can always increase later
  minFrags = 1000,
  addTileMat = TRUE,
  addGeneScoreMat = TRUE
)

## Compute Doublet Scores:
doubScores <- addDoubletScores(
  input = ArrowFiles,
  k = 10, #Refers to how many cells near a "pseudo-doublet" to count.
  knnMethod = "UMAP", #Refers to the embedding to use for nearest neighbor search.
  LSIMethod = 1
)

# add the other arrow files from the pilot:
ArrowFiles <- dir('./')[grepl('.arrow', dir('./'))]

# exclude bad samples:
exclude <- paste0(c('Sample_8', 'Sample_12', 'Sample_19', 'Sample_20', 'Sample_21'), '.arrow')
ArrowFiles <- ArrowFiles[!(ArrowFiles %in% exclude)]

proj <- ArchRProject(
  ArrowFiles = ArrowFiles,
  outputDirectory = "EDA_no_bad",
  copyArrows = TRUE #This is recommened so that you maintain an unaltered copy for later usage.
)

# add sample metadata:
sample_meta <- read.csv('~/swaruplab/smorabit/analysis/PiD_2021/data/sample_meta.csv')
sample_meta$SampleID <- paste0('Sample_', sample_meta$SampleID)

for(meta in names(sample_meta)){
  proj@cellColData[[meta]] <- sample_meta[match(as.character(proj@cellColData$Sample), sample_meta$SampleID), meta]
}

proj <- saveArchRProject(ArchRProj = proj)

```

QC cutoff:

```{r eval=FALSE}

proj <- loadArchRProject(path = "EDA_no_bad")

which(proj$TSSEnrichment > 6 & proj$nFrags > 3000) %>% length

proj<- proj[which(proj$TSSEnrichment > 6 & proj$nFrags > 3000)]
table(proj$Sample)
nrow(proj@cellColData)

```


## Primary Data processing

```{r eval=FALSE}

# LSI
proj <- addIterativeLSI(ArchRProj = proj, useMatrix = "TileMatrix", name = "IterativeLSI", force=TRUE)

# batch correction with Harmony:
proj <- addHarmony(
  proj, reducedDims = "IterativeLSI",
  name = 'Harmony', groupBy='Sample', force=TRUE
)

# Seurat clustering:
proj <- addClusters(input = proj, reducedDims = "Harmony", force=TRUE)

# UMAP:
proj <- addUMAP(ArchRProj = proj, reducedDims = "Harmony", force=TRUE, minDist=0.05)


# plot UMAP by cluster and sample
p1 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Sample", embedding = "UMAP")
p2 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Clusters", embedding = "UMAP")
p3 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "DX", embedding = "UMAP")

plotPDF(p1,p2,p3, name = "qc_filtered-UMAP-Sample-Clusters.pdf",
        ArchRProj = proj, addDOC = FALSE, width = 5, height = 5)


p1 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "nFrags", embedding = "UMAP")
p2 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "TSSEnrichment", embedding = "UMAP")

plotPDF(p1,p2, name = "qc_filtered-UMAP-qc.pdf",
        ArchRProj = proj, addDOC = FALSE, width = 5, height = 5)



```

plot umap with facet wrap:

```{r eval=FALSE}

proj@cellColData$UMAP1 <- proj@embeddings$UMAP$df[,1]
proj@cellColData$UMAP2 <- proj@embeddings$UMAP$df[,2]

p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=Sample)) +
  geom_point(alpha=0.25, size=0.25)

pdf('EDA_no_bad/Plots/umap_samples_split.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()

###############################################################################
# TSS enrichment
###############################################################################


p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=log(TSSEnrichment))) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_tssEnrich.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()

###############################################################################
# Reads in promoter
###############################################################################

p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=log(ReadsInPromoter))) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_ReadsInPromoter.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()

###############################################################################
# Reads in blacklist
###############################################################################

p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=log(ReadsInBlacklist))) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_ReadsInBlacklist.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()

###############################################################################
# nFrags
###############################################################################

p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=log(nFrags))) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_nFrags.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()


###############################################################################
# NucleosomeRatio
###############################################################################

p <- proj@cellColData %>% as.data.frame %>% ggplot(aes(x=UMAP1, y=UMAP2, color=NucleosomeRatio)) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_NucleosomeRatio.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()



```


## Marker gene visualization:

```{r eval=FALSE}

# imputation weights using MAGIC
proj <- addImputeWeights(proj)

canonical_markers <- list(
  'Astrocyte' = c('GFAP', 'AQP4', 'SLC1A2'),
  'Pan-neuronal' = c('SNAP25', 'SYT1'),
  'Excitatory_Neuron' = c('SLC17A7', 'SATB2'),
  'Inhibitory_Neuron' = c('GAD1', 'GAD2'),
  'Microglia' = c('CSF1R', 'CD74', 'P2RY12'),
  'Oligodendrocyte' = c('MOBP', 'MBP', 'MOG'),
  'Olig_Progenitor' = c('PDGFRA', 'CSPG4')
)

# with imputation
p <- plotEmbedding(
    ArchRProj = proj,
    colorBy = "GeneScoreMatrix",
    name = unlist(canonical_markers),
    embedding = "UMAP",
    imputeWeights = getImputeWeights(proj),
    pal=viridis(256)
)
plotPDF(plotList = p,
    name = "Plot-UMAP-Marker-Genes-W-Imputation.pdf",
    ArchRProj = proj,
    addDOC = FALSE, width = 5, height = 5)

# without imputation
p <- plotEmbedding(
    ArchRProj = proj,
    colorBy = "GeneScoreMatrix",
    name = unlist(canonical_markers),
    embedding = "UMAP",
    imputeWeights = NULL,
    pal=viridis(256)
)
plotPDF(plotList = p,
    name = "Plot-UMAP-Marker-Genes-without-Imputation.pdf",
    ArchRProj = proj,
    addDOC = FALSE, width = 5, height = 5)

proj <- saveArchRProject(ArchRProj = proj)


```

Make my own expression plot

```{r eval=FALSE}

getAvailableMatrices(proj)

gene_matrix <- getMatrixFromProject(proj, useMatrix='GeneScoreMatrix')
gm <- assays(gene_matrix)$GeneScoreMatrix
rownames(gm) <- gene_matrix@elementMetadata$name
colnames(gm) <- rownames(proj@cellColData)


# plot GFAP:

df <- proj@cellColData %>% as.data.frame %>% select(c(UMAP1, UMAP2, Sample))
df$expression <- gm['CSF1R',]

p <- df %>% ggplot(aes(x=UMAP1, y=UMAP2, color=log2(expression+1))) +
  geom_point(alpha=0.25, size=0.25) + scale_color_gradientn(colors=rev(magma(256)))
  #geom_hex() + scale_fill_gradientn(colors=rev(magma(256)))

pdf('EDA_no_bad/Plots/umap_gfap.pdf', width=12, height=12)
p + facet_wrap(~Sample) + theme_pubr() +
theme(
  axis.text.x=element_blank(),
  axis.text.y=element_blank(),
  axis.ticks=element_blank(),
)
dev.off()



```

Process one sample only just as a test:

```{r eval=FALSE}


proj_test <- proj[which(proj$Sample == 'Sample_10' & proj$TSSEnrichment > 10)]




# LSI
proj_test <- addIterativeLSI(proj_test, useMatrix = "TileMatrix", name = "IterativeLSI", force=TRUE)

# batch correction with Harmony:
# proj_test <- addHarmony(
#   proj_test, reducedDims = "IterativeLSI",
#   name = 'Harmony', groupBy='Sample'
# )

# Seurat clustering:
proj_test <- addClusters(input = proj_test, reducedDims = "IterativeLSI", force=TRUE)

# UMAP:
proj_test <- addUMAP(ArchRProj = proj_test, reducedDims = "IterativeLSI", force=TRUE, minDist=0.1)


# plot UMAP by cluster and sample
p1 <- plotEmbedding(ArchRProj = proj_test, colorBy = "cellColData", name = "Clusters", embedding = "UMAP")
plotPDF(p1, name = "Sample10-UMAP-Sample-Clusters.pdf",
        ArchRProj = proj, addDOC = FALSE, width = 5, height = 5)




# imputation weights using MAGIC
proj_test <- addImputeWeights(proj_test, force=TRUE)

# with imputation
p <- plotEmbedding(
    proj_test,
    colorBy = "GeneScoreMatrix",
    name = unlist(canonical_markers),
    embedding = "UMAP",
    imputeWeights = getImputeWeights(proj_test),
    pal=viridis(256)
)

plotPDF(plotList = p,
    name = "Sample10-UMAP-Marker-Genes-W-Imputation.pdf",
    ArchRProj = proj_test,
    addDOC = FALSE, width = 5, height = 5)



```
